oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/jackson2/DefaultOidcUserMixin.java

```diff
@@ -43,10 +43,17 @@
 @JsonIgnoreProperties(value = { "attributes" }, ignoreUnknown = true)
 abstract class DefaultOidcUserMixin {
 
+	@Deprecated
 	@JsonCreator
 	DefaultOidcUserMixin(@JsonProperty("authorities") Collection<? extends GrantedAuthority> authorities,
 			@JsonProperty("idToken") OidcIdToken idToken, @JsonProperty("userInfo") OidcUserInfo userInfo,
 			@JsonProperty("nameAttributeKey") String nameAttributeKey) {
 	}
 
+	@JsonCreator
+	DefaultOidcUserMixin(@JsonProperty("name") String name,
```

Comment:
Will you please add `DefaultOAuth2UserMixinTests` and `DefaultOidcUserMixinTests` to ensure that serializations of either arrangement can correctly deserialize?

Reply:
It seems that Jackson only supports a single `@JsonCreator` for each class, meaning the backwards compatibility would be a bit more involved (perhaps using a custom deserializer?).

Reply:
Yes, the next evolution of that mixin would be to use a deserializer.